driver/src/main/java/oracle/nosql/driver/httpclient/HttpClient.java

```diff
@@ -189,33 +189,40 @@ public HttpClient(String host,
         this.maxChunkSize = (maxChunkSize == 0 ?
                               DEFAULT_MAX_CHUNK_SIZE : maxChunkSize);
 
-        workerGroup = new NioEventLoopGroup(numThreads);
+        int cores = Runtime.getRuntime().availableProcessors();
```

Comment:
Javadoc for [availableProcessors() ](https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/lang/Runtime.html#availableProcessors()) points that this method may return different values. This may create a problem with VMs that start with many cores and get reassigned fewer ones down the road. Maybe this could be rechecked once in a while.

Reply:
Once the handle is configured that's it -- the numbers can't be changed without re-creating the handle and Netty config. Netty itself uses that API by default. Users have the ability to choose other numbers.